lib/watcher/cpu-profile.js

```diff
@@ -28,14 +28,14 @@ exports.worker = function(handle) {
       return;
 
     try {
-      var profile = agent().metrics.stopCpuProfiling();
-
-      handle.emit({
-        cmd: 'cpu:profile-data',
-        stalls: stalls,
-        stallout: stallout,
-        profile: profile,
-        // stopTime: Date.now(), XXX could be useful, but doesn't have a user
+      agent().metrics.stopCpuProfiling(function(profile) {
```

Comment:
This code is still wrapped in try/catch. but I believe that it does not throw, so the try/catch should be removed.

The callback structure is unusual, it asynchronous (I assume, because strong-agent uses a direct return, and you needed to change it to a callback). However, the callback is not err-first. This means that this call *cannot* fail? What about when no profile is active? Or... will it throw when no profile is active?

Both throwing exceptions *and* calling back is possible in Node.js, but its atypical, used mostly for severe errors in the construction of the code (args missing or wrong time), not usually for run-time errors.

Net: I think that the callback should be err-first.